pkg/ccl/changefeedccl/changefeed_processors.go

```diff
@@ -422,17 +422,8 @@ func (ca *changeAggregator) setupSpansAndFrontier(
 
 	// Checkpointed spans are spans that were above the highwater mark, and we
 	// must preserve that information in the frontier for future checkpointing.
-	// If we don't have a highwater yet (during initial scan) they must at least
-	// be from StatementTime, and given an initial highwater they must all by
-	// definition have been at or after initialHighWater.Next()
-	var checkpointedSpanTs hlc.Timestamp
-	if initialHighWater.IsEmpty() {
-		checkpointedSpanTs = ca.spec.Feed.StatementTime
-	} else {
-		checkpointedSpanTs = initialHighWater.Next()
-	}
 	for _, checkpointedSpan := range ca.spec.Checkpoint.Spans {
-		if _, err := ca.frontier.Forward(checkpointedSpan, checkpointedSpanTs); err != nil {
+		if _, err := ca.frontier.Forward(checkpointedSpan, ca.spec.Checkpoint.Timestamp); err != nil {
```

Comment:
So, this change (and in particular the code you deleted) is a bit problematic:

```
	var checkpointedSpanTs hlc.Timestamp
	if initialHighWater.IsEmpty() {
		checkpointedSpanTs = ca.spec.Feed.StatementTime
	} else {
		checkpointedSpanTs = initialHighWater.Next()
	}
```

The problem is that you could be consuming the spec checkpoint produced during cluster update, before the new protocol message field was added -- so that checkpoint does not have a timestamp set. 

Perhaps the deleted code can be moved to the distflow where we setup aggregator checkpoints (of course with the comment explaining that that code can be removed in future versions.

Or.... You could just ignore the problem altogether -- no need for code changes.  That effectively means that
checkpoints produced right around cluster upgrade are not good...